packages/lib/src/vis/shared/SelectionTool.tsx

```diff
@@ -17,6 +18,7 @@ interface Props {
   children: (points: Selection) => ReactElement;
 }
 
+// eslint-disable-next-line sonarjs/cognitive-complexity
```

Comment:
As mentioned in the chat, I think we'll be able to remove this by abstracting the drag-with-modifier-key logic.